File: evidence/generated-systems/golang/healthcare-on-golang/healthcare-on-golang/internal/model/ImagingReport.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// ImagingReport Declaration
//==============================================================
type ImagingReport struct {
    gorm.Model
     ReportNumber                                    string
    Impression                                    string
    ReportedDate                                                            time.Time
    ImagingOrderId         *uint
    ImagingOrder           *ImagingOrder `gorm:"foreignKey:ImagingOrderId"`
    ClinicianId         *uint
    Clinician           *Clinician `gorm:"foreignKey:ClinicianId"`
    EncounterId         *uint
    Encounter           *Encounter `gorm:"foreignKey:EncounterId"`
    ImagingCenterId         *uint
    ImagingCenter           *ImagingCenter `gorm:"foreignKey:ImagingCenterId"`
    Status                      ResultStatus

// parent associations as their child

}

File: evidence/generated-systems/angular/healthcare-on-angular/src/app/models/ImagingReport.js

// Define collection and schema for ImagingReport
export interface ImagingReport {
    reportNumber:
	type : string
    impression:
	type : string
    reportedDate:
	type : Date
    ImagingOrder:
	type : Schema.Types.ObjectId
    Clinician:
	type : Schema.Types.ObjectId
    Encounter:
	type : Schema.Types.ObjectId
    ImagingCenter:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'imagingReports'
}
